snorkel/labeling/lf/nlp.py

```diff
@@ -0,0 +1,202 @@
+from typing import Any, Callable, List, Mapping, NamedTuple, Optional
+
+from snorkel.labeling.preprocess import BasePreprocessor
+from snorkel.labeling.preprocess.nlp import EN_CORE_WEB_SM, SpacyPreprocessor
+
+from .core import LabelingFunction
+
+
+class SpacyPreprocessorParameters(NamedTuple):
+    """Parameters need to construct a SpacyPreprocessor."""
+
+    text_field: str
+    doc_field: str
+    language: str
+    disable: Optional[List[str]]
+    preprocessors: List[BasePreprocessor]
+    memoize: bool
+
+
+class SpacyPreprocessorConfig(NamedTuple):
+    """Tuple of SpacyPreprocessor and the parameters used to construct it."""
+
+    nlp: SpacyPreprocessor
+    parameters: SpacyPreprocessorParameters
+
+
+class NLPLabelingFunction(LabelingFunction):
+    """Special labeling function type for SpaCy-based LFs.
+
+    This class is a special version of `LabelingFunction`. It
+    has a `SpacyPreprocessor` integrated which shares a cache
+    with all other `NLPLabelingFunction` instances. This makes
+    it easy to define LFs that have a text input field and have
+    logic written over SpaCy `Doc` objects.
+
+    Simple `NLPLabelingFunction`s can be defined via a
+    decorator. See `nlp_labeling_function`.
+
+    Parameters
+    ----------
+    name
+        Name of the LF
+    f
+        Function that implements the core LF logic
+    resources
+        Labeling resources passed in to `f` via `kwargs`
+    preprocessors
+        Preprocessors to run before SpacyPreprocessor is executed
```

Comment:
What about if you want to run a preprocessor after?  This seems equally likely

Reply:
One could argue this isn't basic usage any more, so might be better served to roll your own using the SpaCy preprocessor. Thoughts?